app/controllers/requests_controller.rb

```diff
@@ -2,14 +2,93 @@ class RequestsController < ApplicationController
   include ApplicationHelper
 
   before_action :logged_in?
+  before_action :has_unreturned_books?, only: :create
 
   def new
+    load_selected_books
+    @request = Request.new
+  end
+
+  def create
+    start_date = request_params[:start_date]
+    end_date = request_params[:end_date]
+    selected_books = request_params[:selected_books]&.map(&:to_i)
+
+    unless valid_borrow_amount? selected_books
+      load_selected_books
+      render :new, status: :unprocessable_entity and return
+    end
+
+    process_request start_date, end_date, selected_books
+  end
+
+  private
+
+  def request_params
+    params.require(:request).permit(Request::PERMITTED_PARAMS)
+  end
+
+  def load_selected_books
     @selected_books = current_user.selected_books
                                   .newest
                                   .includes(book: [:author,
-                                                   {cover_attachment: :blob}])
+                                                  {cover_attachment: :blob}])
                                   .map(&:book)
   end
 
-  def create; end
+  def valid_borrow_amount? selected_books
+    if selected_books.blank?
+      flash.now[:red] = t "error.less_than_a_book"
+    elsif selected_books.size > Settings.request.allow_amount
+      flash.now[:red] = t "error.cant_borrow_more_than",
+                          allow_amount: Settings.request.allow_amount
+    else
+      return true
+    end
+
+    false
```

Comment:
```suggestion
    is_book_present = selected_books.present?
   is_over_size = selected_books.size > Settings.request.allow_amount
   return true if is_book_present && !is_over_size
   
      flash.now[:red] = t "error.less_than_a_book" if !is_book_present
      flash.now[:red] = t("error.cant_borrow_more_than",
                          allow_amount: Settings.request.allow_amount) if is_over_size
    false
```
Viết thử ntn xem có chạy được ko

Reply:
Code clean hơn nhiều rồi ạ, sau em sẽ chú ý khi có nhiều condition thì chuyển nó thành biến cho gọn. Em cảm ơn anh

Reply:
Em hỏi chỗ này một chút ạ.
Vì sao mình lại nên viết theo kiểu dùng flag hả anh, em thấy nếu dùng if-elsif-else thì code sẽ dễ hiểu hơn, với cũng không cần tạo biến local nữa mà :v